src/MediatR.Extensions.Microsoft.AspNetCore/Mediator/RequestAbortedCancellationTokenMediatorDecorator.cs

```diff
@@ -26,34 +29,42 @@ public class RequestAbortedCancellationTokenMediatorDecorator : IMediator
-            return await _mediator.Send(request, cancellationTokenToUse);
+            var cancellationTokenToUse = DecideCancellationTokenUsage(cancellationToken);
+            return await _mediator.Send(request, cancellationTokenToUse).ConfigureAwait(false);
         }
 
-        public async Task<object> Send(object request, CancellationToken cancellationToken = new CancellationToken())
+        public async Task<object?> Send(object request, CancellationToken cancellationToken = default)
         {
-            var cancellationTokenToUse = GetRequestAbortedCancellationTokenOrUseTokenPassedFromInitialCall(cancellationToken);
-            return await _mediator.Send(request, cancellationTokenToUse);
+            var cancellationTokenToUse = DecideCancellationTokenUsage(cancellationToken);
+            return await _mediator.Send(request, cancellationTokenToUse).ConfigureAwait(false);
         }
 
-        public async Task Publish(object notification, CancellationToken cancellationToken = new CancellationToken())
+        public async Task Publish(object notification, CancellationToken cancellationToken = default)
         {
-            var cancellationTokenToUse = GetRequestAbortedCancellationTokenOrUseTokenPassedFromInitialCall(cancellationToken);
-            await _mediator.Publish(notification, cancellationTokenToUse);
+            var cancellationTokenToUse = DecideCancellationTokenUsage(cancellationToken);
+            await _mediator.Publish(notification, cancellationTokenToUse).ConfigureAwait(false);
         }
 
-        public async Task Publish<TNotification>(TNotification notification, CancellationToken cancellationToken = new CancellationToken()) where TNotification : INotification
+        public async Task Publish<TNotification>(TNotification notification, CancellationToken cancellationToken = default) where TNotification : INotification
         {
-            var cancellationTokenToUse = GetRequestAbortedCancellationTokenOrUseTokenPassedFromInitialCall(cancellationToken);
-            await _mediator.Publish(notification, cancellationTokenToUse);
+            var cancellationTokenToUse = DecideCancellationTokenUsage(cancellationToken);
+            await _mediator.Publish(notification, cancellationTokenToUse).ConfigureAwait(false);
         }
 #pragma warning restore 1591
 
-        private CancellationToken GetRequestAbortedCancellationTokenOrUseTokenPassedFromInitialCall(CancellationToken cancellationToken)
+        private CancellationToken DecideCancellationTokenUsage(CancellationToken cancellationToken)
         {
-            return _httpContextAccessor?.HttpContext?.RequestAborted ?? cancellationToken;
+            if (cancellationToken != default && _httpContextAccessor.HttpContext != null)
+            {
+                using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _httpContextAccessor.HttpContext.RequestAborted);
```

Comment:
The CancellationTokenSource will get disposed after exiting the "if" statement, which is before the send/publish command.
Instead, It should be created before the send/publish command and disposed immediately after.
I'm afraid extracting this logic to a method is not an option